mysql-test/include/gap_lock_error_update.inc

```diff
@@ -71,7 +71,6 @@ insert into u1 values (1,2,1,1) on duplicate key update c4=10;
 commit;
 begin;
 select * from u1 where c3=1 and c1 = 1 for update;
---error ER_UNKNOWN_ERROR
 select * from u1 where c3=1 for update;
```

Comment:
OK, here is the real problem, this table, u1, is not created with an engine specifier. I reported/noticed this issue while switching from the custom mtr incantation that specifies default_storage_engine to the standard mtr incantation that does not and dis not make the mental connection of that change over to where this failure started appearing. The test is broken in that this specific table is not properly created with the engine specifier.

That said, looking back at the logic, yes, your original statement is correct, I read through the !using_full_primary_key too fast, which is the first condition to be met for a gap lock. So this change is not the correct fix for the issue.